### DISCUSSION TITLE
Web Sockets On ESP32 Audio Kit

### DISCUSSION THREAD

**Discussion by flashdato** [pschatzmann/arduino-audio-tools#1454] [Q&A] [answered]
is it possible to use the board to play streams from WebSockets? ( I cant find any examples)

this is URL https://dolphin-app-dh42m.ondigitalocean.app/ws

**Comment by pschatzmann** [comment]
I don't quite understand, please elaborate...

**Reply by flashdato** [reply]
instead of using regular links and streaming platforms, I want to send audio data with bits to lower usage 

I though doing with with WebSocket and instead of sending full audio I will send audio bits

**Reply by pschatzmann** [reply, marked as answer]
The short answer is yes - just use a websocket library!

However, I am afraid, it still does not make any sense to me: The WebSocket API is an advanced technology that makes it possible to open a two-way interactive communication session between the user's browser and a server. I don't see how this would make any sense since there is no standardized official audio streaming protocol that you could use and therefore I am not providing any examples.

I also do not understand what you mean with audio bits: All audio is ultimately sent as bitstream. If you want to reduce the data quantity, you will need to use a compressing audio encoder on the sender side and the corresponding decoder on the receiving side.

If you want to send audio data, I suggest that you have a look at the [communication examples](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication).
